### PR TITLE
#1589: add dedicated error message for @static annotation not inside object

### PR DESCRIPTION
**Pull request by esarbe**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 1]
```diff
@@ -1929,4 +1929,11 @@ object messages {
     val explanation =
       hl"An ${"object"} cannot have the same name as an existing ${"package"}. Rename either one of them."
   }
+
+  case class StaticFieldsOnlyAllowedInObjects(field: Symbol)(implicit ctx: Context) extends Message(StaticFieldsOnlyAllowedInObjectsID) {
+    val msg = "@static fields are only allowed inside objects"
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: I would say:
```scala
hl"${"@static"} $member in ${member.owner} must be defined inside an ${"object"}."
```
And use
```scala
hl"${"@static"} members are only allowed inside objects."
```
for the explanation

File: compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 2]
```diff
@@ -1929,4 +1929,11 @@ object messages {
     val explanation =
       hl"An ${"object"} cannot have the same name as an existing ${"package"}. Rename either one of them."
   }
+
+  case class StaticFieldsOnlyAllowedInObjects(field: Symbol)(implicit ctx: Context) extends Message(StaticFieldsOnlyAllowedInObjectsID) {
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: Substitute field by member. It is not necessarily a field. Can be a class, a val...

**Comment by esarbe** [comment]
should be better now

**Comment by allanrenucci** [comment]
Thanks @esarbe! 🎉

**Pull request merged**